### PR TITLE
CC-34100 Add Quicksight client adapter

### PR DESCRIPTION
**Pull request by AsonUnique**
Overview:

- Ticket: https://spryker.atlassian.net/browse/CC-34100

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+name: CI
+
+on:
+    push:
+        branches:
+            - 'master'
+    pull_request:
+    workflow_dispatch:
+
+jobs:
```
Comment (ilyakubanov) [inline .github/workflows/ci.yml:10]: is it a standard list of validations for eco modules? Don't we need any other ones?

File: LICENSE [hunk 2]
```diff
@@ -0,0 +1,40 @@
+SPRYKER SYSTEMS GMBH EVALUATION LICENSE AGREEMENT
```
Comment (ilyakubanov) [inline LICENSE:1]: Don't we need a special License for eco modules?

File: src/SprykerEco/Shared/AmazonQuicksight/AmazonQuicksightConstants.php [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+{
+    /**
+     * Specification:
+     * - The ID for the AWS account that contains your Amazon QuickSight account.
+     *
+     * @api
+     *
+     * @var string
+     */
+    public const AWS_ACCOUNT_ID = 'QUICKSIGHT:AWS_ACCOUNT_ID';
```
Comment (ilyakubanov) [inline src/SprykerEco/Shared/AmazonQuicksight/AmazonQuicksightConstants.php]:
```suggestion
    public const AWS_ACCOUNT_ID = 'AMAZON_QUICKSIGHT:AWS_ACCOUNT_ID';
```
please check everywhere

File: src/SprykerEco/Shared/AmazonQuicksight/AmazonQuicksightConstants.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+    /**
+     * Specification:
+     * - AWS access key ID.
+     *
+     * @api
+     *
+     * @var string
+     */
+    public const AWS_CREDENTIALS_SECRET = 'QUICKSIGHT:AWS_CREDENTIALS_SECRET';
```
Comment (ilyakubanov) [inline src/SprykerEco/Shared/AmazonQuicksight/AmazonQuicksightConstants.php]: specs are duplicated

File: src/SprykerEco/Zed/AmazonQuicksight/Communication/AmazonQuicksightCommunicationFactory.php [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+use Spryker\Zed\Kernel\Communication\AbstractCommunicationFactory;
+
+/**
+ * @method \SprykerEco\Zed\AmazonQuicksight\AmazonQuicksightConfig getConfig()
+ * @method \SprykerEco\Zed\AmazonQuicksight\Persistence\AmazonQuicksightEntityManagerInterface getEntityManager()
+ * @method \SprykerEco\Zed\AmazonQuicksight\Persistence\AmazonQuicksightRepositoryInterface getRepository()
+ * @method \SprykerEco\Zed\AmazonQuicksight\Business\AmazonQuicksightFacadeInterface getFacade()
+ */
+class AmazonQuicksightCommunicationFactory extends AbstractCommunicationFactory
```
Comment (ilyakubanov) [inline src/SprykerEco/Zed/AmazonQuicksight/Communication/AmazonQuicksightCommunicationFactory.php:18]: will we need these empty facade, factories in future?

File: src/SprykerEco/Zed/AmazonQuicksight/Persistence/Propel/Schema/spy_amazon_quicksight.schema.xml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+        <column name="role" required="true" type="VARCHAR" size="255"/>
+        <column name="uuid" required="false" type="VARCHAR" size="255"/>
+
+        <index name="spy_quicksight_user-fk_user">
+            <index-column name="fk_user"/>
+        </index>
+
+        <foreign-key name="spy_quicksight_user-fk_user" foreignTable="spy_user" phpName="User" refPhpName="QuicksightUser">
+            <reference foreign="id_user" local="fk_user"/>
+        </foreign-key>
```
Comment (ilyakubanov) [inline src/SprykerEco/Zed/AmazonQuicksight/Persistence/Propel/Schema/spy_amazon_quicksight.schema.xml:18]: dependency to User module

File: src/SprykerEco/Zed/AmazonQuicksight/Communication/AmazonQuicksightCommunicationFactory.php (hunk 5 above)
Comment (AsonUnique) [reply src/SprykerEco/Zed/AmazonQuicksight/Communication/AmazonQuicksightCommunicationFactory.php:18]: Yes, we will add methods in further stories.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (AsonUnique) [reply .github/workflows/ci.yml:10]: Tbh, I don't know. I guess, we can clarify this closer to the release phase.

File: LICENSE (hunk 2 above)
Comment (AsonUnique) [reply LICENSE:1]: I'll change it to MIT for now, but we'll need to clarify this in the future.

File: src/SprykerEco/Zed/AmazonQuicksight/AmazonQuicksightConfig.php [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+     * Specification:
+     * - Returns the namespace for the Quicksight user registration.
+     *
+     * @api
+     *
+     * @return string
+     */
+    public function getQuicksightUserRegisterNamespace(): string
+    {
+        return 'default';
```
Comment (ilyakubanov) [inline src/SprykerEco/Zed/AmazonQuicksight/AmazonQuicksightConfig.php]: Lets move to constant

File: src/SprykerEco/Zed/AmazonQuicksight/Persistence/Propel/Schema/spy_amazon_quicksight.schema.xml (hunk 6 above)
Comment (ilyakubanov) [reply src/SprykerEco/Zed/AmazonQuicksight/Persistence/Propel/Schema/spy_amazon_quicksight.schema.xml:18]: Not fixed

**Review by ilyakubanov — APPROVED** [review]

**Pull request merged**